src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableDictionary.cs

```diff
@@ -305,7 +305,7 @@ public static class ImmutableDictionary
         ///   <c>true</c> if this map contains the key-value pair; otherwise, <c>false</c>.
         /// </returns>
         [Pure]
-        public static bool Contains<TKey, TValue>(this IImmutableDictionary<TKey, TValue> map, TKey key, TValue value)
+        public static bool Contains<TKey, TValue>(this IImmutableDictionary<TKey, TValue> map, TKey key, TValue value) where TKey : notnull
```

Comment:
If we decide to make IImmutableDictionary not constraint to `notnull` this should probably don't do it either.

Reply:
In that case it also depends on the method implementation right? Like for this method we have `Requires.NotNullAllowStructs(key, nameof(key));` below, so even if IImmutableDictionary allows null we still keep not null for this method right?

Reply:
If the method prohibits null, it's reasonable to keep the constraint.  Though if there are a lot of methods like that, it suggests IImmutableDictionary really wasn't designed to allow null keys.

Reply:
>  Though if there are a lot of methods like that, it suggests IImmutableDictionary really wasn't designed to allow null keys.

So keep non null entire class level too? All methods with key access: Contains(key), Remove(key), Add(key, value), TryGetValue(key, out value), SetItem(key, value) requires non null `key`

Reply:
Removed `TKey : notnull` constraint from `IImmutableDictionary` interface, but kept them for implemented classes `ImmutableDictionary` and `ImmutableSortedDictionary` and corresponding methods as they will throw with null key